include/EncoderDecoder/Huffman/HuffmanEncoderDecoder.h

```diff
@@ -0,0 +1,101 @@
+#ifndef HUFFMAN_ENCODER_DECODER
+#define HUFFMAN_ENCODER_DECODER
+
+#include "BiMap/BiMap.h"
+#include "EncoderDecoder/AbstractEncoderDecoder.h"
+#include "EncoderDecoder/Huffman/PaddedString.h"
+#include "Pair/Hash.h"
+
+#include <algorithm>
+#include <bits/stdc++.h>
+#include <numeric>
+#include <vector>
+
+class HuffmanEncoderDecoder
+	: public AbstractEncoderDecoder<
+		  BiMap<char, std::pair<unsigned char, unsigned int>>,
+		  PaddedString> {
+private:
+	class Node {
+	public:
+		char character;
+		int frequency;
+		std::unique_ptr<Node> leftSon;
+		std::unique_ptr<Node> rightSon;
+
+		Node();
+		Node(char c);
+		Node(Node &&rhs) = default;
+		Node &operator=(Node &&rhs) = default;
+		bool operator==(const Node &rhs) const;
+		Node &operator++(int);
+	};
+
+public:
+	using MappingType = BiMap<char, std::pair<unsigned char, unsigned int>>;
+	using EncodedType = PaddedString;
+	using NodePtr = std::unique_ptr<Node>;
+	using NodeVector = std::vector<NodePtr>;
+	friend class HuffmanEncoderDecoder_MakeEncodingFromTextSubRoutines_Test;
```

Comment:
Maybe move this to the bottom? The idea is to put the class’s interface first and internal stuff later, because the user of the class doesn’t usually care about the internals

Reply:
I use these aliases in the function declaration for convenience. Is that a good enough excuse?

Reply:
I meant only the friend test class

Reply:
TODO